### ISSUE TITLE
Each saved query should reference a specific database

### ISSUE THREAD

**Issue by simonw** [simonw/datasette-saved-queries#3]
Right now we assume that saved queries only execute against the database they were saved in.

If the `saved_queries` table included a `database` column, a single database could be used to save queries against many other (potentially read-only) databases.

As part of this an optional plugin configuration can be used to specify which database the `saved_queries` table should live in, as opposed to now where it uses the first attached database.

**Comment by CharlesNepote** [comment]
I agree. I open my big database in read only mode (`-i`) so I couldn't have a `saved_queries` table in it.
